### DISCUSSION TITLE
Custom Widget interact with default developer portal

### DISCUSSION THREAD

**Discussion by sven5** [Azure/api-management-developer-portal#2606] [Feature request]
Hi,

I'm trying to implement a custom widget for SignIn/SignOut Buttons.
How can I access default design settings, i.e. button types of developer portal from my widget?
For instance, I like to customize the button appearance in my widget to be one of "Normal button" or "Featured button" which are defined in portal style.

**Comment by ghost** [comment]
@sven5, thank you for opening this issue. We will triage it within the next few business days.

**Comment by ghost** [comment]
@sven5, thank you for requesting this feature.

**Comment by JMach1** [comment]
Hello @sven5 

This is not supported as of right now. 

We were thinking about such feature, but we want to prevent issues when we update internal styles, it could break already existing custom widgets and we didn't decided on best approach how to deal with this yet.

The workaround for now is to copy the styles from the portal to CSS files and handle it manually. But I understand it's not ideal.

**Comment by sven5** [comment]
@JMach1 Thanks, it's clear now.
I assume the only way to interact with the developer portal is currently only by using management REST API, right?
